tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -200,6 +202,12 @@ ROOT::TreeUtils::RFriendInfo GetFriendInfo(const TTree &tree, bool retrieveEntri
       const auto *alias_c = tree.GetFriendAlias(frTree);
       const std::string alias = alias_c != nullptr ? alias_c : "";
 
+      auto *treeIndex = frTree->GetTreeIndex();
+      if (treeIndex)
```

Comment:
Maybe it's  also worth checking that `!treeIndex->IsZombie()` since apparently that's done in [BuildIndex](https://github.com/root-project/root/blob/cca7c59a0878ebd7942c6a16516dd34bbb23d912/tree/tree/src/TTree.cxx#L2638) as well

Reply:
ah, I didn't see that, thanks! but I think precisely because `BuildIndex` checks for `isZombie()` on the newly-constructed `TTreeIndex` and does not attach it to the `TTree` if it is then there is no code path where a TTree has a zombie TTreeIndex attached (as it should be).